### PR TITLE
Native aot test app

### PR DESCRIPTION
**Pull request by gregsdennis**
Depends on #624

### CODE REVIEW

File: AotCompatibility/AotCompatibility.csproj [hunk 1]
```diff
@@ -0,0 +1,19 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>net8.0</TargetFramework>
+    <ImplicitUsings>enable</ImplicitUsings>
+    <Nullable>enable</Nullable>
+	<InvariantGlobalization>true</InvariantGlobalization>
+	  
+    <PublishAot>true</PublishAot>
```
Comment (jevansaks) [inline AotCompatibility/AotCompatibility.csproj:10]: Add:
```
		<IsTrimmable>true</IsTrimmable>
```

Comment (gregsdennis) [reply AotCompatibility/AotCompatibility.csproj:10]: Doesn't `PublishAot` cover that?

Comment (gregsdennis) [reply AotCompatibility/AotCompatibility.csproj:10]: https://learn.microsoft.com/en-nz/dotnet/core/deploying/trimming/prepare-libraries-for-trimming?pivots=dotnet-8-0#enable-project-specific-trimming

> The IsTrimmable property defaults to true when configuring a project as AOT-compatible with `<IsAotCompatible>true</IsAotCompatible>`.

Do we need to set that separately from `PublishAot`?

Comment (jevansaks) [reply AotCompatibility/AotCompatibility.csproj:10]: I'm not sure. @sergio0694 always tells me to include it, perhaps he can give better guidance.

Comment (Sergio0694) [reply AotCompatibility/AotCompatibility.csproj:10]: If you use `PublishAot`, that's sufficient. `IsAotCompatible` is used for libraries, but if you're making an app, the former covers everything you need already 🙂

**Comment by gregsdennis** [comment]
Closing this for now.  I think forbidding reflection in the tests is sufficient.

**Pull request closed without merging**